buildchain/buildchain/vagrant.py

```diff
@@ -40,16 +39,60 @@ def mkdir_dot_vagrant() -> None:
         'uptodate': [True],
     }
 
-def task_vagrantup() -> types.TaskDict:
+def task__vagrant_up_noprov() -> types.TaskDict:
+    """Run `vagrant up` without provisioning a development environment."""
+    cmd = [config.VAGRANT, 'up']
+    cmd.extend(map(shlex.quote, config.VAGRANT_UP_ARGS))
+    cmd.append('--no-provision')
+
+    return {
+        'actions': [doit.tools.LongRunning(' '.join(cmd))],
+        'file_dep': [],
+        'task_dep': [],
+        'uptodate': [False],
+    }
+
+def task__vagrant_snapshot() -> types.TaskDict:
+    """Snapshot development environment."""
+    vagranthalt = map(shlex.quote,
+        [config.VAGRANT, 'halt', 'bootstrap'])
+    vagrantsnap = map(shlex.quote,
+        [config.VAGRANT, 'snapshot', 'save', '--force', 'bootstrap',
+         config.VAGRANT_SNAPSHOT_NAME])
+
+    return {
+        'actions': [' '.join(vagranthalt), ' '.join(vagrantsnap)],
```

Comment:
Ha, since you're not using `LongRunning` anymore, you can directly use a list of string instead of a string (and thus no need for `shlex.quote` and `join`.

e.g:

```python
    vagranthalt = [config.VAGRANT, 'halt', 'bootstrap'])
    vagrantsnap = [config.VAGRANT, 'snapshot', 'save', '--force', 'bootstrap',
                   
    return {
        'actions': [vagranthalt, vagrantsnap],
```

Same remark for `task_vagrant_restore`.